segway/gmtk/params.py

```diff
@@ -399,27 +526,71 @@ def __str__(self) -> str:
         return "\n".join(line)
 
     def get_header_info(self) -> str:
+        # No additional header information needed in input.master.
+        return ""
+
+
+class GenericString:
+    """
+    A class storing a generic string to write to input.master.
+    Attributes:
+        contents: str: Generic string to write to input.master
+    """
+    kind = OBJ_KIND_GENERICSTRING
+
+    def __init__(self, contents: str):
+        """
+        Initialize a generic string class.
+        :param contents: generic string to write to input.master
+        """
+        self.contents = contents
+
+    def __str__(self) -> str:
         """
-        No additional header information needed in input.master.
+        Return the stored string.
         """
+        return self.contents
+
+    def get_header_info(self) -> str:
+        # No additional header information
         return ""
 
 
+class DecisionTree(GenericString):
+    """
+    A Decision Tree object.
+    """
+    kind = OBJ_KIND_DT
+
+    def __init__(self, tree: str):
+        """
+        Initialize a DecisionTree object.
+        :param tree: String representation of the tree
+        """
+        super().__init__(tree)
+
+
 class Section(dict):
     """
     Contains GMTK objects of a single type and supports writing them to file.
     Key: name of GMTK object
     Value: GMTK object
     Attributes:
-            kind: str: specifies the kind of GMTK object
-            (default assumes that `self` has no kind)
+            kind: str: specifies the kind of GMTK object (default assumes that
```

Comment:
"default assumes that `self` has no kind" needs more explanation

Reply:
Changed to "default is None".